src/main/java/Duke.java

```diff
@@ -1,10 +1,321 @@
+
+import exceptions.*;
+import todo.*;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.FileNotFoundException;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+//Check if valid command (Done, List, Bye, Action)
+//If done, mark as done
+//If action, check what action is it
+//If there is a date, what is the date
+
 public class Duke {
+
+    protected static ArrayList<Task> tasks = new ArrayList<>();
+    private static ArrayList<String> commands = new ArrayList<>();
+    private static String filePath = "./src/main/java/duke.txt";
     public static void main(String[] args) {
+        String line = "";
+        showWelcomeScreen();
+        addCommands();
+        readFile();
+        while (!line.contains("bye")) {
+            Scanner in = new Scanner(System.in);
+            line = in.nextLine();
+            assert line != null;
+            String[] words = line.split(" ", 2);
+            try{
+                checkValidCommand(words);
+            } catch (InvalidCommandError e) {
+                printDivider();
+                System.out.println("Invalid command. Try again!");
+                printDivider();
+            }
+
+        }
+        return;
+    }
+
+    private static void addCommands() {
+        commands.add("done");
+        commands.add("todo");
+        commands.add("event");
+        commands.add("deadline");
+        commands.add("delete");
+        commands.add("list");
+        commands.add("save");
+        commands.add("bye");
```

Comment:
perhaps consider changing the strings to constants? as you might need these somewhere else?